isabl_cli/app.py

```diff
@@ -1779,7 +1779,7 @@ def validate_species(self, experiments):
         msg = []
 
         for i in experiments:
-            if i["sample"]["individual"]["species"] != self.SPECIES:
+            if self.SPECIES and i["sample"]["individual"]["species"] != self.SPECIES:
```

Comment:
the title of this PR makes me think this should be 
```
if self.ASSEMBLY and i["sample"]["individual"]["species"] != self.SPECIES:
```

Reply:
Yes, that makes sense. You should allow species to be missing when there's no `ASSEMBLY`. Thanks